pkg/cmd/update/update.go

```diff
@@ -0,0 +1,453 @@
+		log.Debug().Msg("skipping update check for --non-interactive command")
+		return
+	}
+
+	tempDir := os.TempDir()
+	tempFilePath := filepath.Join(tempDir, updateStatusFileName)
+	file, err := os.OpenFile(tempFilePath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		log.Warn().Err(err).Msgf("could not open file to set update status %s", tempFilePath)
+		return
+	}
+	defer file.Close()
+
+	data, err := os.ReadFile(tempFilePath)
+	if err != nil {
+		log.Warn().Err(err).Msgf("could not read timestamp from update status file: %s", tempFilePath)
+		return
+	}
+
+	now := time.Now()
+	previousTimeString := string(data)
+	previousTime, err := time.Parse(time.RFC3339, previousTimeString)
+	// This is expected to fail on first run
+	if err != nil {
+		log.Debug().Err(err).Msgf("could not parse timestamp from update status file: %s previousTimeString: \"%s\"", tempFilePath, previousTimeString)
+		// go's time.Sub only works with time.Time, not time.Duration
+		previousTime = now.Add(-updateInterval)
+	}
+
+	timeSince := now.Sub(previousTime)
+	if timeSince >= updateInterval {
+		githubToken := cfg.GitHub.Token.Value
+
+		// Update the previousTime since we're checking
+		_, err := file.WriteString(now.Format(time.RFC3339))
+		if err != nil {
+			log.Warn().Err(err).Msgf("could not write timestamp to update status file: %s", tempFilePath)
+			return
+		}
+		err = file.Sync()
+		if err != nil {
+			log.Warn().Err(err).Msgf("could not sync update status file: %s", tempFilePath)
+			return
+		}
+
+		githubClient := github.NewClient(nil)
+		if githubToken != "" {
+			githubClient = githubClient.WithAuthToken(githubToken)
+		}
+		available, version, err := updateAvailable(githubClient)
+		if err != nil {
+			log.Warn().Err(err).Msg("could not check for updates")
+			return
+		}
+
+		if available {
+			styles := userio.NewNonInteractiveStyles()
+			fmt.Println(
+				styles.Bold.Inherit(styles.InfoStyle).
+					Render(fmt.Sprintf("corectl %s is available, run `corectl update` to install.", version)),
+			)
+		}
+	} else {
+		timeLeft := (updateInterval - timeSince).Round(time.Second)
+		log.Debug().Msgf("next update check will be in %s", timeLeft)
+	}
+}
+
+func getCurrentExecutablePath() string {
+	execPath, err := os.Executable()
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to get executable path")
+	}
+
+	absolutePath, err := filepath.Abs(execPath)
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to get absolute executable path")
+	}
+
+	return absolutePath
+}
+
+func UpdateCmd(cfg *config.Config) *cobra.Command {
+	opts := UpdateOpts{
+		githubToken:   "",
+		targetVersion: "",
+	}
+	updateCmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update corectl",
+		Long:  `Update to the latest corectl version.`,
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.targetVersion = args[0]
+			}
+			if cfg.IsPersisted() {
+				opts.githubToken = cfg.GitHub.Token.Value
```

Comment:
Let's add a flag for github token and not check if token is persisted

Reply:
the token is optional for the release endpoints, I'll remove the IsPersisted though